test/utils.py

```diff
@@ -21,6 +21,22 @@ def needs_cuda(test_item):
     return test_item
 
 
+def cpu_and_cuda():
+    # TODO: Figure out how to dedupe this logic in needs_cuda() and here.
+    if not torch.cuda.is_available():
+        if os.environ.get("FAIL_WITHOUT_CUDA") == "1":
+            raise RuntimeError("CUDA is required for this test")
+        return ("cpu",)
+    return ("cpu", pytest.param("cuda", marks=pytest.mark.needs_cuda))
+
+
+def get_tensor_compare_function(device):
```

Comment:
I'm a bit uncomfortable with naming this `get_tensor_compare_function` because it should be frame-specific, not tensor specific. E.g. for tensors like `pts_seconds`, we'd want to use a stricter equality check both on CPU and GPU.

Basically I think this should be `get_frame_compare_function()`.